server.js

```diff
@@ -41,17 +41,18 @@ app.post('/process-payment', async (req, res) => {
   const request_params = req.body;
 
   // length of idempotency_key should be less than 45
-  const idempotency_key = crypto.randomBytes(22).toString('hex');
+  const idempotency_key = request_params.idempotency_key;
```

Comment:
nit: not sure the preference here, but if we're creating consts for idempotency_key, would we also want to for nonce and location_id just to be consistent?